tensorflow_quantum/python/optimizers/rotosolve_minimizer_test.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2020 The TensorFlow Quantum Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Test module for tfq.python.optimizers.rotosolve_minimizer optimizer."""
+from operator import mul
+from functools import reduce
+import numpy as np
+import tensorflow as tf
+from absl.testing import parameterized
+import cirq
+import sympy
+from tensorflow_quantum.python.layers.high_level import pqc
+from tensorflow_quantum.python import util
+from tensorflow_quantum.python.optimizers import rotosolve_minimizer
+
+
+def loss_function_with_model_parameters(model, loss, train_x, train_y):
+    """Create a new function that assign the model parameter to the model
+    and evaluate its value.
+
+    Args:
+        model : an instance of `tf.keras.Model` or its subclasses.
+        loss : a function with signature loss_value = loss(pred_y, true_y).
+        train_x : the input part of training data.
+        train_y : the output part of training data.
+
+    Returns:
+        A function that has a signature of:
+            loss_value = f(model_parameters).
+    """
+
+    # obtain the shapes of all trainable parameters in the model
+    shapes = tf.shape_n(model.trainable_variables)
+    count = 0
+    sizes = []
+
+    # Record the shape of each parameter
+    for shape in shapes:
+        n = reduce(mul, shape)
+        sizes.append(n)
+        count += n
+
+    # Function accept the parameter and evaluate model
+    @tf.function
+    def func(params):
+        """A function that can be used by tfq.optimizer.rotosolve_minimize.
+
+        Args:
+           params [in]: a 1D tf.Tensor.
+
+        Returns:
+            Loss function value
+        """
+
+        # update the parameters of the model
+        start = 0
+        for i, size in enumerate(sizes):
+            model.trainable_variables[i].assign(
+                tf.reshape(params[start:start + size], shape))
+            start += size
+
+        # evaluate the loss
+        loss_value = loss(model(train_x, training=True), train_y)
+        return loss_value
+
+    return func
+
+
+class RotosolveMinimizerTest(tf.test.TestCase, parameterized.TestCase):

```

Comment:
We are going to need a few more tests here than just what is in the docstring snippet. Keep in mind also that the docstring snippet and the test code will end up looking slightly different since in tests we use things like `util.convert_to_tensor` which for a docstring snippet would become `tfq.convert_to_tensor`.

I would suggest maybe some simpler tests that don't need to make use `tf.keras.Models` which test more atomic behavior and make sure you are getting the right shapes and output values in various cases. Maybe also some cases that verify a hand calculated minimum/optimum value. Also make sure that tests contain `self.Assert****` type statements in them so we know that things are working. Right now, this test could break and we might never know it :)

Reply:
Thanks very much. I have added a new simple test for the optimum value. Sorry, it was a mistake for forgetting `self.Assert****`, I have put them all back. Many thanks